src/values.rs

```diff
@@ -285,27 +383,22 @@ pub enum Value {
     Json(Arc<Json>),
 
     /// Binary data.
-    Binary {
-        /// The actual data. We put it behind an `Arc` to make sure
-        /// that cloning remains inexpensive.
-        data: Arc<Vec<u8>>,
-        mimetype: String
-    }
+    Binary(Binary),
 }
 
 impl Value {
     pub fn get_type(&self) -> Type {
         match *self {
             Value::Unit => Type::Unit,
-            Value::OnOff => Type::OnOff,
-            Value::OpenClosed => Type::OpenClosed,
+            Value::OnOff(_) => Type::OnOff,
```

Comment:
Well, since you define `Value:OnOff(OnOff)`, I believe(d) that you need to also pattern-match against `Value::OnOff(_)`. I have no idea why it worked without the `(_)`, but at some point, a test complained that it didn't work.